packages/cli/src/commands/__tests__/deploy.test.ts

```diff
@@ -0,0 +1,77 @@
+import { afterAll, afterEach, beforeAll, expect, vi } from 'vitest'
+import { decidePackageName, deploy } from '../deploy'
+import { getPersonalAccessToken } from '../../utils'
+
+vi.mock('../../utils')
+
+vi.mock('../deploy', async () => {
+  const actualDeploy = await vi.importActual<typeof import('../deploy')>(
+    '../deploy',
+  )
+
+  const mocks = await vi.importMock<typeof import('../deploy')>('../deploy')
+
+  return {
+    ...mocks,
+    decidePackageName: vi.fn(),
+    deploy: actualDeploy.deploy,
+  }
+})
+
+describe('deploy', () => {
+  afterEach(() => {
+    vi.resetAllMocks()
+  })
+
+  afterAll(() => {
+    vi.unstubAllGlobals()
+  })
+
+  it('should exit when getPersonalAccessToken returns an error', async () => {
+    const exit = vi.fn(() => {
```

Comment:
currently, I am creating the stub in each of the two tests as it did not work and the stub was not reset or caught by the secondly-run test so it always failed